optimum/graphcore/ipu_configuration.py

```diff
@@ -178,6 +209,74 @@ def __init__(self, **kwargs):
                 matmul_proportion = 0.2
             setattr(self, f"{mode}_matmul_proportion", kwargs.pop(f"{mode}_matmul_proportion", matmul_proportion))
 
+            # mode based layer configurations
+            # serialized_linear_splits_per_ipu and serialized_embedding_splits_per_ipu defaults
+            # depend on the model architecture
+            setattr(
+                self,
+                f"{mode}_embedding_serialization_factor",
+                kwargs.pop(f"{mode}_embedding_serialization_factor", embedding_serialization_factor),
+            )
+            setattr(
+                self,
+                f"{mode}_linear_serialization_factor",
+                kwargs.pop(f"{mode}_linear_serialization_factor", linear_serialization_factor),
+            )
+            setattr(
+                self,
+                f"{mode}_serialized_embedding_splits_per_ipu",
+                kwargs.pop(f"{mode}_serialized_embedding_splits_per_ipu", serialized_embedding_splits_per_ipu),
+            )
+            setattr(
+                self,
+                f"{mode}_serialized_linear_splits_per_ipu",
+                kwargs.pop(f"{mode}_serialized_linear_splits_per_ipu", serialized_linear_splits_per_ipu),
+            )
+
+            # validate serialized_{layer}_splits_per_ipu if provided
+            mode_ipus_per_replica = getattr(self, f"{mode}_ipus_per_replica")
+            for layer in ("embedding", "linear"):
+                mode_layer_config = f"{mode}_serialized_{layer}_splits_per_ipu"
+                mode_layer_splits_per_ipu = getattr(self, mode_layer_config)
+                mode_layer_serialization_factor = getattr(self, f"{mode}_{layer}_serialization_factor")
+                if mode_layer_splits_per_ipu is not None:
+                    # serialized_{layer}_splits_per_ipu must be of type List[int>=0]
+                    if (
+                        not isinstance(mode_layer_splits_per_ipu, list)
+                        or not all(isinstance(x, int) for x in mode_layer_splits_per_ipu)
+                        or any(x < 0 for x in mode_layer_splits_per_ipu)
+                        or sum(mode_layer_splits_per_ipu) < 1
```

Comment:
This might just be personal preference but I find this version of the condition easier to read:
```suggestion
                    if not (
                        isinstance(mode_layer_splits_per_ipu, list)
                        and all(isinstance(x, int) for x in mode_layer_splits_per_ipu)
                        and all(x >= 0 for x in mode_layer_splits_per_ipu)
                        and sum(mode_layer_splits_per_ipu) > 0
```